csharp/src/Apache.Arrow/Ipc/ArrowReaderImplementation.cs

```diff
@@ -165,8 +165,21 @@ internal static ByteBuffer CreateByteBuffer(ReadOnlyMemory<byte> buffer)
                 throw new InvalidDataException("Null count length must be >= 0"); // TODO:Localize exception message
             }
 
-            ArrowBuffer[] arrowBuff = new[] { nullArrowBuffer, valueArrowBuffer };
-            ArrayData[] children = GetChildren(ref recordBatchEnumerator, field, bodyData);
+            ArrowBuffer[] arrowBuff = null;
```

Comment:
No need to initialize these 2 variables to `null` since they will always be set in the `if-else`.

1. It is a minor perf improvement.
2. It allows the compiler to warn you in case you didn't set them.